File: testsuite/basic/src/test/java/io/smallrye/faulttolerance/reuse/sync/MyFaultTolerance.java
package io.smallrye.faulttolerance.reuse.sync;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;

import io.smallrye.common.annotation.Identifier;
import io.smallrye.faulttolerance.api.FaultTolerance;

@ApplicationScoped
public class MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    public static final FaultTolerance<String> FT = FaultTolerance.<String> create()
            .withRetry().maxRetries(2).done()
            .withFallback().handler(() -> "fallback").done()
            .build();
}

File: testsuite/basic/src/test/kotlin/io/smallrye/faulttolerance/kotlin/reuse/MyFaultTolerance.kt
package io.smallrye.faulttolerance.kotlin.reuse

import io.smallrye.common.annotation.Identifier
import io.smallrye.faulttolerance.api.FaultTolerance
import java.util.function.Supplier
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Produces

@ApplicationScoped
object MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    val FT = FaultTolerance.create<String>()
            .withRetry().maxRetries(2).done()
            .withFallback().handler(Supplier { "fallback" }).done()
            .build()
}
